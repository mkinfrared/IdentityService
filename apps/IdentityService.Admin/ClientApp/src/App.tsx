import { Spinner, useTheme } from "@identity-service/ui";
import { Suspense } from "react";
import { BrowserRouter } from "react-router-dom";

import AppRoutes from "routes/AppRoutes";

import css from "./App.module.scss";

const App = () => {
  useTheme("app-theme");

  return (
    <BrowserRouter>
      <div className={css.App}>
        <Suspense
          fallback={<Spinner className={css.spinner} type="infinity" />}
        >
          <AppRoutes />
        </Suspense>
      </div>
    </BrowserRouter>
  );
};

export default App;
