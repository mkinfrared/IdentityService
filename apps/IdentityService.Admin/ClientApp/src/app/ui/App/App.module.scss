@use "shared/styles/common" as c;

.App {
  :where(&) {
    --background-light: var(--white);
    --background-dark: var(--black-light);
    --background: var(--background-dark);
  }

  align-items: center;
  background-color: var(--background);
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 100%;
  min-height: 100vh;
  overflow-x: hidden;

  @include c.theme("light") {
    --background: var(--background-light);
  }

  .spinner {
    left: 50%;
    position: fixed;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 33%;

    path {
      stroke: var(--primary);
    }
  }

  .content {
    display: flex;
    flex: 1;
    width: 100%;
  }
}
