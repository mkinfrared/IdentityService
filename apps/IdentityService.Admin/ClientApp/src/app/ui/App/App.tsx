import { Spin } from "antd";
import { Suspense } from "react";

import { useUserSession } from "shared/lib/hooks";
import { Header } from "widgets/header";
import { Main } from "widgets/main";
import { MainSidebar } from "widgets/mainSidebar";

import { AppRouterProvider, withProviders } from "../../providers";

import css from "./App.module.scss";

const App = () => {
  const user = useUserSession();

  return (
    <div className={css.App}>
      <Header />
      <div className={css.content}>
        <Suspense fallback={<Spin className={css.spinner} size="large" />}>
          {user && <MainSidebar className={css.sidebar} />}
          <Main className={css.main}>
            <AppRouterProvider />
          </Main>
        </Suspense>
      </div>
    </div>
  );
};

export default withProviders(App);
