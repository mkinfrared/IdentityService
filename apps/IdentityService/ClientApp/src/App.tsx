import { QueryClientProvider } from "react-query";

import Main from "pages/Main";
import queryClient from "utils/queryClient";

const App = () => (
  <QueryClientProvider client={queryClient}>
    <Main />
  </QueryClientProvider>
);

export default App;
