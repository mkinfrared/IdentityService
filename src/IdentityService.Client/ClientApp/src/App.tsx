import { BrowserRouter } from "react-router-dom";

import "./App.scss";

const App = () => (
  <BrowserRouter>
    <div className="App" />
  </BrowserRouter>
);

export default App;
