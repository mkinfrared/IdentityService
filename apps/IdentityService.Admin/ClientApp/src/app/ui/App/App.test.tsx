import { App } from "./App";

jest.mock("routes/AppRoutes");

jest.mock("@identity-service/ui");

describe("<App />", () => {
  it("should be defined", () => {
    expect(App).toBeDefined();
  });
});
