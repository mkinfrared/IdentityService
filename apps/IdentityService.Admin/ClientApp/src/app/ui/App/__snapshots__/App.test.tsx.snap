// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="Header"
      data-testid="Header"
    >
      Header works!!
    </div>
    <div
      class="content"
    >
      <div
        aria-busy="true"
        aria-live="polite"
        class="ant-spin ant-spin-lg ant-spin-spinning spinner css-dev-only-do-not-override-1m62vyb"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
